metadata-ingestion/tests/integration/powerbi/test_powerbi.py

```diff
@@ -1505,3 +1505,52 @@ def test_independent_datasets_extraction(
         output_path=tmp_path / "powerbi_independent_mces.json",
         golden_path=f"{test_resources_dir}/{golden_file}",
     )
+
+
+@freeze_time(FROZEN_TIME)
+@mock.patch("msal.ConfidentialClientApplication", side_effect=mock_msal_cca)
```

Comment:
Just a thought, Instead of golden files, can we write unit tests for` bigquery.Mapper.extract_lineage`. Definitely repeating a lot of unrelated mcps across multiple goldens files.

Reply:
butter to have this file as it completely verify from Pipeline. Solution is to have bytebuffer MCP writer to avoid golden file approach.

Reply:
I think we need to add a bit more unit testing on the CLL, but it's also helpful to have a full test of this sort even though it repeats a lot of MCPs

Reply:
Updated the test `test_sqlglot_parser` for CLL